pkgs/development/ocaml-modules/ppx_deriving_cmdliner/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, buildDunePackage
+, fetchFromGitHub
+, alcotest
+, cmdliner
+, cppo
+, ppx_deriving
+, ppx_tools
+, ppxfind
+, ppxlib
+}:
+
+buildDunePackage rec {
+  pname = "ppx_deriving_cmdliner";
+  version = "0.5.1";
+
+  minimumOCamlVersion = "4.03";
+  useDune2 = true;
+
+  src = fetchFromGitHub {
+    owner = "hammerlab";
+    repo = "ppx_deriving_cmdliner";
+    rev = "v${version}";
+    sha256 = "1sszcxjay6p1glkw9riih6w7hl1isa81gyhi2gx1sp5qrqps51di";
+  };
+
+  nativeBuildInputs = [
+    cppo
+    ppxfind
+  ];
+
+  propagatedBuildInputs = [
+    cmdliner
+    ppx_deriving
+    ppx_tools
+    ppxlib
```

Comment:
ppx_tools and ppxlib need not be propagated.

Reply:
`ppx_tools` is not needed anymore in the new version [0.6.0](https://github.com/hammerlab/ppx_deriving_cmdliner/releases/tag/v0.6.0).

It seems that `ppxlib` still needs to be propagated. Otherwise one has to explicitly mention it when compiling a test package which depends on `ppx_deriving_cmdliner`.

Reply:
That’s true. Things would be cleaner if the runtime and the ppx were packaged separately (but that’s probably for an other day).